src/mlpack/methods/ann/image_functions/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+# Define the files we need to compile
+# Anything not in this list will not be compiled into mlpack.
+set(SOURCES
+  bilinear_impl.hpp
```

Comment:
Moreover, you didn't add `image_functions` dir into `src/mlpack/methods/ann/CMakeLists.txt`.